apps/ledger-live-desktop/src/renderer/modals/Receive/Body.tsx

```diff
@@ -197,6 +222,23 @@ const Body = ({
         setTitle(t("receive.title"));
     }
   }, [steps, stepId, t, currency.name]);
+  useEffect(() => {
+    async function computeAccountError() {
+      const addressSanctioned = await checkAddressBlacklisted(account);
+      if (addressSanctioned && account.type === "Account") {
+        setAccountError(
+          new AddressesSanctionedError("AddressesSanctionedError", {
+            addresses: [account.freshAddress],
+          }),
+        );
+      } else {
+        const error = account ? getReceiveFlowError(account, parentAccount) : undefined;
+        setAccountError(error);
+      }
+    }
```

Comment:
We could define this function outside the `useEffect` and use it in the `handleChangeAccount` too as they are the same, wdyt ?